### PR TITLE
Adds documentation for kubernetes list stateful sets

### PR DESCRIPTION
**Pull request by m-orfanos**
### Fixes [MC-10754](https://cloud-ops.atlassian.net/browse/MC-10754)

See https://github.com/cloudops/cloudmc-kubernetes-sdk/pull/51
See https://github.com/cloudops/cloudmc-kubernetes-plugin/pull/23

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,4 @@ build/
 _yardoc
 doc/
 .idea/
+vendor
```
Comment (m-orfanos) [inline .gitignore:25]: when running `bundle exec middleman server` locally to view the files, the tool creates this folder, so I added it to the gitignore

**Review by Gerudhoh — COMMENTED** [review]
I have no idea what I'm looking at, hahaha sorry for the useless review

**Review by garrettkinman — APPROVED** [review]
I presume that not all the JSON attributes being listed in the table is because they're self-explanatory and/or of less importance? It all looks good to me, then.

File: source/includes/gcp/_k8_statefulsets.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+<code>GET /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/statefulsets</code>
+
+Retrieve a list of all stateful sets in a given [environment](#administration-environments).
+
+| Attributes                                 | &nbsp;                                                                |
+| ------------------------------------------ | --------------------------------------------------------------------- |
+| `id` <br/>*string*                         | The id of the stateful set                                            |
+| `metadata` <br/>*object*                   | The metadata of the the stateful set                                  |
+| `metadata.creationTimestamp` <br/>*string* | The date of creation of the stateful set as a string                  |
+| `metadata.labels` <br/>*map*               | The labels associated to the stateful set and there associated values |
```
Comment (Gerudhoh) [inline source/includes/gcp/_k8_statefulsets.md]: The wording here is a little awkward to me. The  way that the word "associated" is repeated does not sit with me, a particularly pedantic English minor. So, you know, grain of salt...

Also, use remember to use the possessive "their", not "there" here and on subsequent lines ;)

File: source/includes/kubernetes/_k8_statefulsets.md [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+| `metadata.creationTimestamp` <br/>*string* | The date of creation of the stateful set as a string                  |
+| `metadata.labels` <br/>*map*               | The labels associated to the stateful set and there associated values |
+| `metadata.name` <br/>*string*              | The name of the stateful set                                          |
+| `metadata.namespace` <br/>*string*         | The namespace in which the stateful set is created                    |
+| `metadata.uid` <br/>*object*               | The UUID of the stateful set                                          |
+| `spec`<br/>*object*                        | The specification used to create and run the stateful set             |
+| `status`<br/>*object*                      | The status information of the the stateful set                        |
+| `status.readyReplicas`<br/>*integer*       | Total number of ready pods targeted by this deployment                |
+| `status.replicas`<br/>*integer*            | Total number of non-terminated pods targeted by this deployment       |
+
```
Comment (Gerudhoh) [inline source/includes/kubernetes/_k8_statefulsets.md]: Why is the same info repeated 4 times? Just curious.

**Review by Gerudhoh — CHANGES_REQUESTED** [review]

File: source/includes/gcp/_k8_statefulsets.md (hunk 2 above)
Comment (m-orfanos) [reply source/includes/gcp/_k8_statefulsets.md]: nice catch!

File: source/includes/kubernetes/_k8_statefulsets.md (hunk 3 above)
Comment (m-orfanos) [reply source/includes/kubernetes/_k8_statefulsets.md]: The doc needs to exist under the gcp plugin and the k8s standalone plugin. Then in each of those there's `/get` and /get/id` (so 2x2=4). The attributes are the same throughout.

**Review by garrettkinman — CHANGES_REQUESTED** [review]
See Julia's requested changes. Good catch on her part with the "there/their".

File: source/includes/gcp/_k8_statefulsets.md (hunk 2 above)
Comment (m-orfanos) [reply source/includes/gcp/_k8_statefulsets.md]: I removed the "and..." part. I copy-pasted it from the official docs without proofreading 🤦  8a67d85acf842bb3127d200f31526fd28cf7a834

Comment (m-orfanos) [reply source/includes/gcp/_k8_statefulsets.md]: Even worse... it wasn't the official docs. I copy-pasted from another file :p 

I fixed them all here 2ae8873e6165bc3851499878a98e8ead9c7decc5

File: source/includes/gcp/_k8_pods.md [hunk 4]
```diff
@@ -212,15 +212,15 @@ Retrieve a list of all pods in a given [environment](#administration-environment
...
+`cluster_id` <br/>*string* | The id of the cluster in which to list the pods.
 
 Attributes | &nbsp;
 ------- | -----------
-`id` <br/>*string* | The id of the pod.  
+`id` <br/>*string* | The id of the pod.
 `metadata` <br/>*object* | The metadata of the the pod
 `metadata.annotations` <br/>*map* | The annotations of the pod
 `metadata.creationTimestamp` <br/>*string* | The date of creation of the pod as a string
-`metadata.labels` <br/>*map* | The labels associated to the pod and there associated values
```
Comment (Gerudhoh) [inline source/includes/gcp/_k8_pods.md:223]: It pains me that we're doing away with the punctuation but oh well :p This looks great Milt! I love how you solved both problems with one delete, ahha.

**Review by Gerudhoh — APPROVED** [review]

File: source/includes/kubernetes/_k8_statefulsets.md (hunk 3 above)
Comment (Gerudhoh) [reply source/includes/kubernetes/_k8_statefulsets.md]: Thanks for clarifying!

**Review by garrettkinman — APPROVED** [review]

**Review by y0geshdev — APPROVED** [review]
LGTM.

File: source/includes/kubernetes/_k8_statefulsets.md [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+| ------------------------------------------ | --------------------------------------------------------------- |
+| `id` <br/>*string*                         | The id of the stateful set                                      |
+| `metadata` <br/>*object*                   | The metadata of the the stateful set                            |
+| `metadata.creationTimestamp` <br/>*string* | The date of creation of the stateful set as a string            |
+| `metadata.labels` <br/>*map*               | The labels associated to the stateful set                       |
+| `metadata.name` <br/>*string*              | The name of the stateful set                                    |
+| `metadata.namespace` <br/>*string*         | The namespace in which the stateful set is created              |
+| `metadata.uid` <br/>*object*               | The UUID of the stateful set                                    |
+| `spec`<br/>*object*                        | The specification used to create and run the stateful set       |
+| `status`<br/>*object*                      | The status information of the the stateful set                  |
```
Comment (y0geshdev) [inline source/includes/kubernetes/_k8_statefulsets.md]: Additional `the` can be removed.

File: source/includes/kubernetes/_k8_statefulsets.md [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+| ------------------------------------------ | --------------------------------------------------------------- |
+| `id` <br/>*string*                         | The id of the stateful set                                      |
+| `metadata` <br/>*object*                   | The metadata of the the stateful set                            |
+| `metadata.creationTimestamp` <br/>*string* | The date of creation of the stateful set as a string            |
+| `metadata.labels` <br/>*map*               | The labels associated to the stateful                           |
+| `metadata.name` <br/>*string*              | The name of the stateful set                                    |
+| `metadata.namespace` <br/>*string*         | The namespace in which the stateful set is created              |
+| `metadata.uid` <br/>*object*               | The UUID of the stateful set                                    |
+| `spec`<br/>*object*                        | The specification used to create and run the stateful set       |
+| `status`<br/>*object*                      | The status information of the the stateful set                  |
```
Comment (y0geshdev) [inline source/includes/kubernetes/_k8_statefulsets.md]: Same as above.

Comment (m-orfanos) [reply source/includes/kubernetes/_k8_statefulsets.md]: e56d9f4fa4cc2bdb50c51822c11b30f5af7f81fd

File: source/includes/kubernetes/_k8_statefulsets.md (hunk 5 above)
Comment (m-orfanos) [reply source/includes/kubernetes/_k8_statefulsets.md]: e56d9f4fa4cc2bdb50c51822c11b30f5af7f81fd

**Comment by y0geshdev** [comment]
Thanks for the changes @m-orfanos . I also removed the additional `the` from Pods docs as I too was copy-pasting it from there 😅 😅

**Comment by m-orfanos** [comment]
> Thanks for the changes @m-orfanos . I also removed the additional `the` from Pods docs as I too was copy-pasting it from there 😅 😅

There are several more :p f8b278850498698e31055e9bc5b2821f594fca8c

**Pull request merged**